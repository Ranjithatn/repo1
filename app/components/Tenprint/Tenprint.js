import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";

import WorkflowWrapper from "../WorkflowWrapper/WorkflowWrapper";
import LeftContainer from "../WorkflowWrapper/LeftContainer";
import Button from "../Button/Button";
import Dropdown from "../Dropdown/Dropdown";
import DropdownItem from "../Dropdown/DropdownItem";
import Title from "../Title/Title";
import Subtitle from "../Title/Subtitle";

import "./Tenprint.scss";
import {requestStartScanType} from "../../actions/jobs"
import { liveScanSelector } from "../../selectors/liveScan";
import { pageSelector } from "../../selectors/jobs";
import Permissions from '../Permissions';
import { liveScanWorkflow } from "../../actions/liveScan";


export class Tenprint extends React.Component {
  constructor() {
    super();
  }

  render() {
    const { formatMessage, history, liveScan, page,requestStartScanType, liveScanWorkflow } = this.props;

    return (

      <Fragment>
        <div className="main-top">
          <Button
            text={formatMessage({ id: "cancel" })}
            className="is-primary"
            onClick={() => {
              history.push("/authenticated/jobqueue");
              if (page === "scanned") {
                history.push("/authenticated/ScannedBiometrics/Scanned");
              } else {
                history.push("/authenticated/jobqueue");
              }

              // history.goBack()

              // if ( liveScan.status === "RESET" ) {
              // } else {
              //   history.push("/authenticated/jobqueue");
              // }
            }}
          />
          <Title is="4" text={formatMessage({ id: "title.tenprintInput" })} />
        </div>
        <div className="main-content">
          <div className="tenprint-selection">
            <Permissions version={2} type="menu.criminal.enroll">
              <Button
                className="is-primary is-medium scancard-btn"
                // onClick={() => history.push("/authenticated/tenprint/card")}
                onClick={() => requestStartScanType("Cardscan")}
                text={formatMessage({ id: "scanCard" })}
                leftIcon="print"
              />
            </Permissions>
            &nbsp;&nbsp;&nbsp;
            <Permissions version={2} type="menu.criminal.enroll">
              <Button
                className="is-primary is-medium"
                // onClick={() => history.push("/authenticated/tenprint/livescan")}
                onClick={() => {
                  liveScanWorkflow({type: "RESET"});
                  requestStartScanType("Livescan")
                } }
                text={formatMessage({ id: "liveScan" })}
                leftIcon="hand-paper-o"
              />
            </Permissions>
          </div>
          <Subtitle
            is="5"
            text={formatMessage({ id: "msg.chooseTenprintInput" })}
          />
        </div>
      </Fragment>
    );
  }
  // render ends here
}

// export default withRouter(Tenprint);
const mapState = state => ({
  liveScan: liveScanSelector(state),
  page: pageSelector(state)
});

export default connect(mapState, {requestStartScanType, liveScanWorkflow})(withRouter(Tenprint));
