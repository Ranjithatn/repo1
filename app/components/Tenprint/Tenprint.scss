@import "../../colors.scss";
#tenprint {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  .main-top {
    display: flex;
    align-items: center;
    justify-content: enter;
    padding: 10px;
    background-color: $dark-gray;
    min-height: 45px;
    button {
      position: absolute;
    }
    .title {
      color: $white;
      margin-right: auto;
      margin-left: auto;
    }
  }
  .main-content {
    flex-direction: column;
    display: flex;
    flex-grow: 1;
    background-color: $dark;
    padding-top: 8%;
    overflow: auto;
    .tenprint-selection {
      margin-right: auto;
      margin-left: auto;
    }
    .subtitle {
      margin-top: 10px;
      margin-right: auto;
      margin-left: auto;
      color: $white;
    }
  }
}
// .Workflow-Wrapper.tenprint-main {
//   .left {
//     // .left-heading {
//     //   display: flex;
//     //   align-items: center;
//     //   padding: 0 .75rem 0 .75rem;
//     //   height: 80px;
//     //   font-size: 26px;
//     // }
//     .tenprint-format-container {
//       display: flex;
//       flex-direction: column;
//       align-items: center;
//       flex-grow: 1;
//       .dropdown-trigger>button {
//         background: transparent;
//         border: none;
//         color: white;
//       }
//       .dropdown-item {
//         text-align: center;
//         &.is-hoverable:hover {
//           background: $green;
//           color: $white;
//           cursor: pointer;
//         }
//       }
//     }
//     .img-format-container {
//       padding: 5px;
//       display: flex;
//       background: $dark;
//       color: $white;
//       font-size: 15px;
//     }
//     .img-container {
//       height: 90px;
//       width: 90px;
//       background: $white;
//     }
//   }
//   .main {
//     .main-container {
//       .msg {
//         .subtitle {
//           color: $white;
//         }
//         text-align: center;
//       }
//       .buttons {
//         display: flex;
//         justify-content: center;
//         padding-top: 5rem;
//       }
//       .cardscan-link {
//         padding-right: 15%;
//       }
//     }
//   }
// }