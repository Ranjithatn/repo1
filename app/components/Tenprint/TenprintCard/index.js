import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import CardscanSettings from "components/CardscanSettings";

import Button from "../../Button/Button";
import Title from "../../Title/Title";
import "./TenprintCard.scss";
import { pageSelector } from "../../../selectors/jobs";
class TenprintCard extends Component<null, null> {
  render() {
    const { formatMessage, history, page } = this.props;
    return (
      <Fragment>
        <div className="main-top">
          <Button
            text={formatMessage({ id: "back" })}
            className="is-primary"
            // onClick={() => history.push("/authenticated/tenprint")}
            onClick={() => {
            
              if (page === "scanned") {
                history.push("/authenticated/ScannedBiometrics/Scanned");
              } else {
                history.push("/authenticated/tenprint");
              }

              // history.goBack()

              // if ( liveScan.status === "RESET" ) {
              // } else {
              //   history.push("/authenticated/jobqueue");
              // }
            }}
          />
          <Title is="4" text={formatMessage({ id: "title.tenprintInput" })} />
        </div>
        <CardscanSettings
          formatMessage={formatMessage}
          onCompleted={() =>
            history.push("/authenticated/tenprint/card/scanned")
          }
        />
      </Fragment>
    );
  }
}

const mapState = state => ({
  page: pageSelector(state)
});

export default connect(
  mapState,
  {}
)(withRouter(TenprintCard));
