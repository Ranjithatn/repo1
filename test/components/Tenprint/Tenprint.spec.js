import React from "react";
import toJson from "enzyme-to-json";
import { Tenprint } from "../../../app/components/Tenprint/Tenprint";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

const props = {
  formatMessage: data => data.id
}

describe('TenprintVerifyLeft', () => {

  xit("should render without crashing", () => {
    const component = mount(
      <Tenprint {...props} />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

});
