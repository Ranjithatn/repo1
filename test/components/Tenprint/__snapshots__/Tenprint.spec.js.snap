// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenprintVerifyLeft should render without crashing 1`] = `
<Tenprint
  formatMessage={[Function]}
>
  <div
    className="main-top"
  >
    <Button
      className="is-primary"
      onClick={[Function]}
      text="cancel"
    >
      <button
        className="button is-primary"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          cancel
        </span>
      </button>
    </Button>
    <Title
      as="h1"
      is="4"
      text="title.tenprintInput"
    >
      <h1
        className="title is-4"
      >
        title.tenprintInput
      </h1>
    </Title>
  </div>
  <div
    className="main-content"
  >
    <div
      className="tenprint-selection"
    >
      <Button
        className="is-primary is-medium scancard-btn"
        leftIcon="print"
        onClick={[Function]}
        text="scanCard"
      >
        <button
          className="button is-primary is-medium scancard-btn"
          disabled={false}
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          <Icon
            icon="print"
          >
            <span
              className="icon"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              title=""
            >
              <i
                className="fa fa-print"
              />
            </span>
          </Icon>
          <span>
            scanCard
          </span>
        </button>
      </Button>
         
      <Button
        className="is-primary is-medium"
        leftIcon="hand-paper-o"
        onClick={[Function]}
        text="liveScan"
      >
        <button
          className="button is-primary is-medium"
          disabled={false}
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          <Icon
            icon="hand-paper-o"
          >
            <span
              className="icon"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              title=""
            >
              <i
                className="fa fa-hand-paper-o"
              />
            </span>
          </Icon>
          <span>
            liveScan
          </span>
        </button>
      </Button>
    </div>
    <Subtitle
      as="h1"
      is="5"
      text="msg.chooseTenprintInput"
    >
      <h1
        className="subtitle is-5"
      >
        msg.chooseTenprintInput
      </h1>
    </Subtitle>
  </div>
</Tenprint>
`;
